### PR TITLE
icon-lang: fix build

### PR DESCRIPTION
**Pull request by sugar700**
###### Motivation for this change

ZHF: #80379

@NixOS/nixos-release-managers

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xfix** [comment]
@GrahamcOfBorg build icon-lang noweb

**Comment by xfix** [comment]
`noweb` fails, but this seems unrelated.

**Comment by xfix** [comment]
@GrahamcOfBorg build icon-lang noweb

I have no idea what's the difference, but that seems to work.

**Comment by xfix** [comment]
@GrahamcOfBorg build icon-lang noweb

**Comment by yurrriq** [comment]
Seems good to me. Should we do a separate PR to fix installTargets?

**Review by yurrriq — APPROVED** [review]

File: pkgs/development/interpreters/icon-lang/default.nix [hunk 1]
```diff
@@ -12,6 +12,11 @@ stdenv.mkDerivation rec {
 
   buildInputs = stdenv.lib.optionals withGraphics [ libX11 libXt ];
 
+  patches = [
+    # Adapted from https://github.com/gtownsend/icon/commit/bfc4a6004d0d3984c8066289b8d8e563640c4ddd
```
Comment (worldofpeace) [inline pkgs/development/interpreters/icon-lang/default.nix]: Did you try this?
```
patches = [
  (fetchpatch {
    url = "https://github.com/gtownsend/icon/commit/bfc4a6004d0d3984c8066289b8d8e563640c4ddd";
    sha256 = ...
    excludes = [
      "doc/relnotes.htm"
      "src/h/version.h"
    ];
  })
];
```

Hopefully the `#  Contributor:  Cheyenne Wills` isn't going to be an issue :crossed_fingers:

**Comment by xfix** [comment]
@GrahamcOfBorg build icon-lang noweb

**Review by yurrriq — APPROVED** [review]
LGTM

**Review by vcunat — APPROVED** [review]

**Comment by KoviRobi** [comment]
Hmm, I know bash quoting is notoriously weird, but I think the change to `noweb` isn't right, let me think out loud (apologies, it isn't meant to be patronising, all the steps surprised me apart from the first one)

- `"cmd ${var[@]}"` does `cmd "${var[0]}" "${var[1]" ... "${var[n]}"`, so I can see why `make "''${installFlags[@]}" install-man` (note, in a doubly single-quoted string in nix, hence the `''${...}`) would make sense.
- I had no idea what `"prefix ${var[@]} suffix"` does, so I checked (in bash):
  ```
  print_args() {
      for a in "$@"; do
          echo $a
      done
  }
  typeset -a var=(1 2 3 4)
  print_args "hello ${var[@]} world"
  ```
  gives
  ```
  hello 1
  2
  3
  4 world
  ```

  So doing make "''${installFlags[@]} install-man" wouldn't make sense, because it would be like
  ```
  make "VAR1=VAL1" "VAR2=VAL2" ... "VARN=VALN install-man"
  ```
  I.e. it would just make the default target! Except, it gets even worse:
- Turns out nix passes lists as space-separated values, so `installFlags` isn't an array, it's a single string:
  ```
  print_args "${installFlags[@]} install-man"
  ```
  gives (I have used `${placeholder out}` instead of `$out` hence the root-relative hashes)
  ```
  BIN=/1rz4g4znpzjwh1xymhjpm42vipw92pr73vdgl6xs1hycac8kf2n9/bin ELISP=/1rz4g4znpzjwh1xymhjpm42vipw92pr73vdgl6xs1hycac8kf2n9/share/emacs/site-lisp LIB=/1rz4g4znpzjwh1xymhjpm42vipw92pr73vdgl6xs1hycac8kf2n9/lib/noweb MAN=/1rz4g4znpzjwh1xymhjpm42vipw92pr73vdgl6xs1hycac8kf2n9/share/man TEXINPUTS=/0wi6yxl11zryxm8g3s6iqz0karl3k4bc3zpvma0ci713wzid8ixw/tex/latex/noweb install-man
  ```
  so `make` only receives the variable `BIN` and it has the value `/1rz... ELISP=... LIB=... MAN=... TEXINPUTS=... install-man`, ouch! This certainly isn't what I expected.

  Which is how I got to my change of (on top of your change here):
  ```diff
  @@ -57,7 +57,7 @@ let noweb = stdenv.mkDerivation rec {
   
       # HACK: This is ugly, but functional.
       PATH=$out/bin:$PATH make -BC xdoc
  -    make "''${installFlags[@]} install-man"
  +    make ''${installFlags[@]} install-man
   
       ln -s "$tex" "$out/share/texmf"
     '';
  ```
  Granted, this will break if you have a space in any single element of `installFlags` :( I wonder if this is something to be changed in the way nix passes lists to derivations?

**Pull request merged**